src/main/kube-auth-proxy/create-kube-auth-proxy-cert-files.injectable.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { writeFile } from "fs/promises";
+import { ensureDir } from "fs-extra";
+import directoryForUserDataInjectable from "../../common/app-paths/directory-for-user-data/directory-for-user-data.injectable";
+import { getKubeAuthProxyCertificatePath } from "./kube-auth-proxy-cert";
+import * as selfsigned from "selfsigned";
+import { createKubeAuthProxyCertFiles } from "./create-kube-auth-proxy-cert-files";
+
+const createKubeAuthProxyCertFilesInjectable = getInjectable({
+  id: "create-kube-auth-proxy-cert-files",
+
+  instantiate: async (di) => {
+    const userData = di.inject(directoryForUserDataInjectable);
+    const certPath = getKubeAuthProxyCertificatePath(userData);
+    
+    return createKubeAuthProxyCertFiles(certPath, {
```

Comment:
Given that this creates the files and then returns `certPath` (which coincendentally should probably be called something like `directoryForProxyCertificates`. I think that this injectable should actually be mostly just `setup` and then instantiate should return `undefined` or maybe a new `SetupOnly` marker type.

Then whatever depends on `createKubeAuthProxyCertFilesInjectable` could instead just depend on `directoryForProxyCertificatesInjectable` and know that the setup and creation of the files will already have been done.